### PR TITLE
Feature/alexar bridge

### PR DESCRIPTION
**Pull request by yrong**
- provisioning scripts and contracts for integration with alexar bridge
- substrate call with evm signature

https://www.notion.so/parallelfinance/Alexar-Bridge-f8c521b02a6c446cb899662fd9750efb

### CODE REVIEW

**Review by GopherJ — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -18,10 +18,11 @@ CUMULUS_DOCKER_TAG									:= v0.9.24
 init: submodules
 	git config advice.ignoredHook false
 	git config core.hooksPath .githooks
-	curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y --default-toolchain nightly-2022-05-11 --component rust-src --component rustfmt --component clippy --target wasm32-unknown-unknown
+	curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y --default-toolchain nightly-2022-07-24 --component rust-src --component rustfmt --component clippy --target wasm32-unknown-unknown
 	cargo install cargo-udeps --locked
 	cd scripts/helper && yarn
 	cd scripts/polkadot-launch && yarn
+	cd scripts/evm && npm install
```
Comment (GopherJ) [inline Makefile:25]: prefer yarn

File: Makefile [hunk 2]
```diff
@@ -115,10 +120,18 @@ integration-test-kusama-call:
 integration-test-sibling-transfer:
 	RUST_LOG="xcm=trace,xcm-executor=trace" SKIP_WASM_BUILD= cargo test -p runtime-integration-tests -- polkadot_transfer --nocapture
 
+.PHONY: test-pallet-evm-precompile-assets-erc20
```
Comment (GopherJ) [inline Makefile:123]: consistency is important

`test-evm...` directly is better because we implies already pallet in `test-prices` etc

File: Makefile [hunk 3]
```diff
@@ -115,10 +120,18 @@ integration-test-kusama-call:
...
+test-pallet-evm-precompile-assets-erc20:
+	SKIP_WASM_BUILD= cargo test -p pallet-evm-precompile-assets-erc20 --lib --no-fail-fast -- --nocapture
```
Comment (GopherJ) [inline Makefile:125]: same as above

File: Makefile [hunk 4]
```diff
@@ -115,10 +120,18 @@ integration-test-kusama-call:
...
+
 .PHONY: test-pallet-evm-precompile-balances-erc20
 test-pallet-evm-precompile-balances-erc20:
 	SKIP_WASM_BUILD= cargo test -p pallet-evm-precompile-balances-erc20 --lib --no-fail-fast -- --nocapture
 
+.PHONY: test-pallet-evm-signatures
```
Comment (GopherJ) [inline Makefile:131]: same as above

File: Makefile [hunk 5]
```diff
@@ -240,9 +253,17 @@ dev-launch: shutdown
...
-.PHONY: local-dev-launch
-local-dev-launch:
-	cargo run --locked --bin parallel --features with-evm-runtime --features runtime-benchmarks --features try-runtime -- --tmp --alice --dev --rpc-cors all --unsafe-ws-external --unsafe-rpc-external --ws-port 19944 --rpc-port 29933
+.PHONY: launch-evm
+launch-evm:
+	cargo run --locked --bin parallel --features with-evm-runtime --features runtime-benchmarks --features try-runtime -- --tmp --alice --dev --rpc-cors all --unsafe-ws-external --rpc-methods unsafe --unsafe-rpc-external --ws-port 19944 --rpc-port 29933
+
+.PHONY: provisioning-evm
+provisioning-evm:
+	cd scripts/evm && npm run build && npm run init-chain && npm run provisioning-chain
```
Comment (GopherJ) [inline Makefile:262]: prefer yarn

File: runtime/kerria/src/lib.rs [hunk 6]
```diff
@@ -297,6 +299,7 @@ impl Contains<Call> for BaseCallFilter {
             || matches!(
                 call,
                 // System, Currencies
+                Call::System(_) |
```
Comment (GopherJ) [inline runtime/kerria/src/lib.rs:302]: it's already whitelisted

File: runtime/vanilla/src/lib.rs [hunk 7]
```diff
@@ -299,6 +301,7 @@ impl Contains<Call> for BaseCallFilter {
             || matches!(
                 call,
                 // System, Currencies
+                Call::System(_) |
```
Comment (GopherJ) [inline runtime/vanilla/src/lib.rs:304]: already whitelisted

File: runtime/vanilla/src/lib.rs [hunk 8]
```diff
@@ -1054,6 +1058,31 @@ impl pallet_ethereum::Config for Runtime {
     type StateRoot = pallet_ethereum::IntermediateStateRoot<Self>;
 }
 
+parameter_types! {
+    pub const EcdsaUnsignedPriority: TransactionPriority = TransactionPriority::MAX / 2;
+    pub const CallFee: Balance = 1 * DOLLARS / 10;
+    pub const CallMagicNumber: u16 = 1592;
```
Comment (GopherJ) [inline runtime/vanilla/src/lib.rs:1064]: is this somehow related to EVMChainId?

File: docs/EVM.md [hunk 9]
```diff
@@ -0,0 +1,19 @@
+# Public Endpoints
+
+|Network |Type  | Endpoint| GasLimit
+--- | --- | ---| ---
+|kerria-test-net|JSON RPC|https://kerria-rpc.parallel.fi:29933| 15_000_000
```
Comment (GopherJ) [inline docs/EVM.md:5]: shall we use standard eth rpc port?

File: runtime/vanilla/src/lib.rs (hunk 8 above)
Comment (yrong) [reply runtime/vanilla/src/lib.rs]: yes, same. Some refactor accordingly

File: docs/EVM.md (hunk 9 above)
Comment (yrong) [reply docs/EVM.md:5]: guess this public endpoint can be configured by DevOps without exposing port,  and for local test sometime conflict cause trouble
